cmd/app/serve.go

```diff
@@ -80,6 +80,7 @@ func newServeCmd() *cobra.Command {
 	cmd.Flags().String("grpc-host", "0.0.0.0", "The host on which to serve requests for GRPC")
 	cmd.Flags().String("grpc-port", "8081", "The port on which to serve requests for GRPC")
 	cmd.Flags().String("metrics-port", "2112", "The port on which to serve prometheus metrics endpoint")
+	cmd.Flags().Duration("read-header-timeout", 60*time.Second, "The time allowed to read the headers of the requests in seconds")
```

Comment:
Is this too long, should it be ~10 seconds or less (not much in the headers besides an auth token)? I assume we have request timeouts that would trigger before this, though I'm not sure where those are configured (k8s?).

Reply:
nginx uses 60s by default
i will reduce that to 10

Reply:
done

we need to check what is the value we are setting on the nginx side (`client_header_timeout`)